lib/SILGen/SILGenPattern.cpp

```diff
@@ -2317,7 +2317,13 @@ void PatternMatchEmission::emitCaseBody(CaseStmt *caseBlock) {
 
   // Implicitly break out of the pattern match statement.
   if (SGF.B.hasValidInsertionPoint()) {
-    SGF.emitBreakOutOf(CleanupLocation(caseBlock), PatternMatchStmt);
+    // Case blocks without trailing braces have ambiguous cleanup locations.
+    SILLocation cleanupLoc = getCompilerGeneratedLocation();
+    if (auto *braces = dyn_cast<BraceStmt>(caseBlock->getBody()))
+      if (braces->getNumElements() == 1 &&
+          dyn_cast_or_null<DoStmt>(braces->getElement(0).dyn_cast<Stmt *>()))
```

Comment:
`&& isa<DoStmt>(...)`

Reply:
Ah I see.. you need an isa_or_null.

Reply:
Thanks! isa_or_null would be better, but I'd rather not force the bots to re-test this patch. I've added this to my list of yaks.